### PR TITLE
Move crates to crates directory

### PR DESCRIPTION
**Pull request by Stupremee**
This PR moves all crates, except `xtask` to a `crates` directory.

I hope I replaced all paths correctly.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,14 +1,14 @@
...
-  "rslint_parser",
-  "rslint_core",
-  "rslint_cli",
-  "rslint_lsp",
+  "crates/rslint_syntax",
+  "crates/rslint_lexer",
+  "crates/rslint_parser",
+  "crates/rslint_core",
+  "crates/rslint_cli",
+  "crates/rslint_lsp",
```
Comment (RDambrosio016) [inline Cargo.toml:8]:
```suggestion
  "crates/*"
```

File: xtask/src/ast.rs [hunk 2]
```diff
@@ -817,7 +817,6 @@ pub(crate) const AST_SRC: AstSrc = AstSrc {
 
         struct ClassDecl {
             T![class],
-            name: Name,
```
Comment (RDambrosio016) [inline xtask/src/ast.rs:820]: 🤔

File: xtask/src/docgen/mod.rs [hunk 3]
```diff
@@ -134,7 +134,7 @@ pub fn group_markdown(data: &[(String, RuleFile)], group: &Group) -> String {
         ));
     }
     ret.push_str(&format!(
-        "\n[Source](../../../rslint_core/src/groups/{})",
+        "\n[Source](../../../crates/rslint_core/src/groups/{})",
```
Comment (RDambrosio016) [inline xtask/src/docgen/mod.rs:137]: You must rerun cargo docgen to update all of the doc files

**Review by RDambrosio016 — COMMENTED** [review]
Looks good to me except for a few nits 👍

File: xtask/src/ast.rs (hunk 2 above)
Comment (Stupremee) [reply xtask/src/ast.rs:820]: Oh whoops

**Review by RDambrosio016 — APPROVED** [review]
LGTM

**Pull request merged**